pandas/core/internals/blocks.py

```diff
@@ -2583,7 +2567,7 @@ def _block_shape(values: ArrayLike, ndim: int = 1) -> ArrayLike:
     return values
 
 
-def safe_reshape(arr, new_shape: Shape):
+def safe_reshape(arr: ArrayLike, new_shape: Shape) -> ArrayLike:
```

Comment:
The docstring implies that Series and Index are allowed

Reply:
updated the docstring